android/festago/app/src/test/java/com/festago/festago/presentation/ui/ticketreserve/TicketReserveViewModelTest.kt

```diff
@@ -0,0 +1,84 @@
+package com.festago.festago.presentation.ui.ticketreserve
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.festago.festago.domain.model.Reservation
+import com.festago.festago.domain.repository.ReservationRepository
+import com.festago.festago.presentation.model.ReservationUiModel
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.setMain
+import org.assertj.core.api.AssertionsForClassTypes.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class TicketReserveViewModelTest {
+    private lateinit var vm: TicketReserveViewModel
+    private lateinit var reservationRepository: ReservationRepository
+
+    private val fakeReservation = Reservation(0)
+
+    @get:Rule
+    val instantExecutorRule = InstantTaskExecutorRule()
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(UnconfinedTestDispatcher())
+        reservationRepository = mockk()
+        vm = TicketReserveViewModel(reservationRepository)
+    }
+
+    @Test
+    fun `예약 정보를 불러오면 성공 이벤트가 발생하고 리스트를 반환한다`() {
+        // given
+        coEvery {
+            reservationRepository.loadReservation()
+        } answers {
+            Result.success(List(10) { fakeReservation })
```

Comment:
같은 예약 정보의 리스트를 뱉기보다 더 일반적인 상황이 좋을 것 같아요.

```suggestion
            Result.success((1..10).map{ Reservation(it) } )
```

많이 쓰인다면 이를 위한 fixture 나 팩토리 메서드를 만드는 것도 좋다고 생각합니다.

Reply:
좋은 의견인거 같습니다!👍
다만 가짜 객체라는 것을 알리기 위해서 제안주신 것을`fakeReservations`로 명명하는게 좋을 것같네요!